anndata/compat/__init__.py

```diff
@@ -62,12 +62,37 @@ class Literal(metaclass=LiteralMeta):
             pass
 
 
+def _read_hdf5_attribute(attrs: h5py.AttributeManager, name: str):
+    """
+    Read an HDF5 attribute and perform all necessary conversions.
+
+    At the moment, this only implements conversions for string attributes, other types
+    are passed through. String conversion is needed compatibility with other languages.
+    For example Julia's HDF5.jl writes string attributes as fixed-size strings, which
+    are read as bytes by h5py.
+    """
+    attr = attrs[name]
+    attr_id = attrs.get_id(name)
+    dtype = h5py.check_string_dtype(attr_id.dtype)
+    if dtype is None:
+        return attr
+    else:
+        if dtype.length is None:  # variable-length string, no problem
+            return attr
+        elif len(attr_id.shape) == 0:  # Python bytestring
+            return attr.decode("utf-8")
+        else:  # NumPy array
+            return attr.astype("U")
+
+
 def _from_fixed_length_strings(value):
     """\
     Convert from fixed length strings to unicode.
 
     For backwards compatability with older h5ad and zarr files.
     """
+    if not hasattr(value, "dtype"):
+        return value
```

Comment:
What case is this for? My understanding is that Julia will write `Array{String, N}` as variable length strings.

My general concern here is that I would rather not touch the backwards compat stuff unless we need to.

Reply:
Ah, apologies, I think I was trying to use that function at some point and forgot to remove my changes. I'll remove this.